Change font of date in recipient row and date separators.

amanagr opened the pull request:
discussion: https://chat.zulip.org/#narrow/stream/101-design/topic/date.20separators

Contains commits from #23099 and #22661 with an additional commit for deploying to chat.zulip.org.
before:
<img width="616" alt="Screenshot 2022-11-29 at 4 40 31 PM" src="https://user-images.githubusercontent.com/25124304/204514420-691ba8e4-8fc7-4c68-9a50-0646620bd54a.png">
<img width="616" alt="Screenshot 2022-11-29 at 4 40 20 PM" src="https://user-images.githubusercontent.com/25124304/204514434-08c56080-c062-403e-9223-d3f8cee0149b.png">

after:
<img width="616" alt="Screenshot 2022-11-29 at 4 36 29 PM" src="https://user-images.githubusercontent.com/25124304/204514022-b4e0166e-4778-4ab8-ab61-e5586a2ce42b.png">
<img width="616" alt="Screenshot 2022-11-29 at 4 41 31 PM" src="https://user-images.githubusercontent.com/25124304/204514401-f797f548-37cc-4297-88c2-6bb47bcfbd58.png">

terpimost commented:
**Suggestions**

Lets do the time font-weight 300
<img width="61" alt="image" src="https://user-images.githubusercontent.com/1903309/204965916-de1f0f5b-9cab-47f0-8f28-2d5bbc3f5264.png">

For the date and day lets do `font-weight: 500;`
<img width="76" alt="image" src="https://user-images.githubusercontent.com/1903309/204966002-1d135818-8802-4db6-8927-1ce16f74a941.png">

For the line separator we should remove one border so the line is actually 1 px:
<img width="1544" alt="image" src="https://user-images.githubusercontent.com/1903309/204966183-9a6d220e-3b0d-45c5-aca3-04d11e21328e.png">

Don't forget to update the color of the line to 
"20% black for the light theme and 50% white for the dark theme, but we might tune those later after some usage
<img width="421" alt="image" src="https://user-images.githubusercontent.com/1903309/204966466-4db08eba-5051-4986-87ab-0af50790583f.png">

amanagr commented:
Pushed to update some old code since we moved from `body` to `:root` to toggle dark theme.

timabbott commented:
Thanks for keeping pushing on this!

This looks slightly below center in terms of vertical alignment:

![image](https://user-images.githubusercontent.com/2746074/205184525-73049ca1-3385-4a36-8a4e-515887edc6df.png)

Looks fine at 125% zoom, in case that's relevant. My guess is we've got some silly super manual padding CSS that we need to replace there.

More importantly, with my 1920x1080 external monitor (which is a relatively low resolution / old monitor), the times are unreadably faint. (I realize this is one of those annoying things where it looks different on different screens). I think this is probably a blocker for test-deploying this.

Also I noticed we're using 60% opacity in the date dividers and 75% in the recipient bars, and 50% in the timestamps; is this configuration intended? 

I played around a bit and here's what it'd look like if we did 
* 75% opacity for the date text, with the current fonts in this PR
* 60% opacity for the time text, with normal font weight and settings (no small caps), 1em sizing.

![image](https://user-images.githubusercontent.com/2746074/205185016-ff00978a-235b-40f6-b5c0-ac06d1eebbca.png)

I think the timestamps are a lot more in the space I'd be hoping for in terms of being super easy to read (and also more similar looking to other content in the app).

I'm realizing while writing this that I think the small caps look OK in the date headings; it's mainly in the timestamps where they bother me. I think I'd prefer to not use the small caps font at all for download size reasons if we can get a similar look without doing so, but I don't feel strongly on that and it sounds like it's already been discussed.

As a broader comment: Obviously, if needed we can use a lot different fonts in order to make the UI look good, we can do that. But from a maintainability standpoint, I would be happier if we had a strategy for how we're using fonts that we can write down somewhere in the codebase, and at least I can understand and enforce when reviewing changes. 

We don't necessarily need to have these immediately while doing this redesign, but If we can start defining a menu of specific font configurations (color, weight, settings, ) that we intend to use in redesign components, that we can use in some sort of consistent fashion, that would make integrating changes like this feel a lot easier, since we can just pick the best fitting item from the menu (or decide none are suitable, and add a new one to the menu), but when reading the CSS, I don't need to be asking myself "Did we really intend to use this novel combination of font color+opacity+weight, or was it probably a transcription error?" when reviewing something, and I'll be better able to help fix anomalies when I see them.

alya commented:
@terpimost Just making sure you se @timabbott 's comment above.

amanagr commented:
Updated with requested changes:

> I played around a bit and here's what it'd look like if we did
> 
> 75% opacity for the date text, with the current fonts in this PR
> 60% opacity for the time text, with normal font weight and settings (no small caps), 1em sizing.

<img width="65" alt="image" src="https://user-images.githubusercontent.com/25124304/205558677-a04f8b3f-9810-48d5-845b-ced06d2a377f.png">

The opacity changes seemed intentional coming from #23099, I removed them and applied the above.

amanagr commented:
I changed the message time back to 12px based on feedback that it was too big in this PR.

timabbott commented:
The size looks good, but it doesn't look to be centered properly inside message rows that don't have a sender heading:

![image](https://user-images.githubusercontent.com/2746074/213303041-842aa951-da7a-4098-88bd-e276156706f9.png)

Please test this sort of alignment detail in a range of scenarios when making adjustments to message layout CSS -- it's the kind of thing that's 50% likely to be broken if not explicitly tested when tweaking something.

(I think this is otherwise ready to test-deploy again).

timabbott commented:
I played around with this a bit more, and noticed this quirk in dark theme: 

![image](https://user-images.githubusercontent.com/2746074/213305786-5657853d-4008-422b-aedf-7d2473ef348b.png)

The color of the dates in the recipient bars is more white than in the divider rows (the white `color-default-text` used for text in message bodies, not the more grey default font color for the font theme -- with both of these being before the 75% opacity discussed above is applied). Testing, it looked better to both Alya and me for us to change the recipient bars to match the font color for the timestamps and date divider rows, than the other way around, at least in the dark theme. But probably a similar consistency issue exists in the light theme, and I think it's probably a good principle that we should just have the dates in that column all use the same font details.

@terpimost thoughts?

amanagr commented:
Fixed the time alignment in messages.

timabbott commented:
The vertical alignment of times still seem to be off at some zoom levels (this is 120% in Chrome on Linux):

![image](https://user-images.githubusercontent.com/2746074/213599029-b8eadc77-96d6-4594-874f-36b073cf9a65.png)

I feel like the `top: 10px` model is kinda guaranteed to have this kind of artifact -- I wonder if we'd be better off doing some sort of vertical alignment CSS for this component instead.

timabbott commented:
Test-deploying this on chat.zulip.org.

timabbott commented:
I'm hoping we can integrate this soon, since I think the floating recipient bar changes PR merge conflicts with this one, and I'd like to be testing that on chat.zulip.org too.

timabbott commented:
Has this been updated with Vlad's latest design suggestion?

amanagr commented:
Waiting on response from @terpimost in the czo discussion.

amanagr commented:
> I think we do want date separators to be selectable, but only in the redesigned state where we just show the new date in them. So that can be tweaked in the PR for that project.

terpimost commented:
Guys, I suggest to do this 
<img width="909" alt="image" src="https://user-images.githubusercontent.com/1903309/218646802-fd7d952b-f65e-4e8e-9994-3a9cbdc61f24.png"> otherwise numbers look ugly.
But if you feel differently go ahead.

amanagr commented:
Updated!

terpimost commented:
looks good to me

amanagr commented:
Pushed to fix failing CI.

timabbott commented on static/styles/dark_theme.css:
```diff
@@ -577,6 +577,10 @@
 
+    .message_row .date_row span {
+        opacity: 0.75;
+    }
```
I removed this likely stale hunk, which seemed to introduce an accidental 0.75 squared opacity (and made this color different from the one selected via `color-date` below).

timabbott commented:
Merged with the small change noted here: https://github.com/zulip/zulip/pull/23696/commits/eb9e0e3b91a877b9717123a948a2e2aa8e4fab9f#r1116322417

Thanks so much for all the work on this piece of the redesign @amanagr @terpimost and @JoeyPriceless!
